addons/mrp/models/mrp_unbuild.py

```diff
@@ -119,12 +119,19 @@ def action_unbuild(self):
         for produce_move in produce_moves:
             if produce_move.has_tracking != 'none':
                 original = original_quants.filtered(lambda quant: quant.product_id == produce_move.product_id)
-                self.env['stock.move.lots'].create({
-                    'move_id': produce_move.id,
-                    'lot_id': original.lot_id.id,
-                    'quantity_done': produce_move.product_uom_qty,
-                    'quantity': produce_move.product_uom_qty
-                })
+                group_by_lot = {}
+                for o in original:
+                    if o.lot_id.id in group_by_lot:
```

Comment:
You can replace these 4 lines by
`group_by_lot.setdefault(o.lot_id.id, 0) += produce_move.product_uom_qty`